swagger.yaml

```diff
@@ -107,265 +243,926 @@ paths:
+    get:
+      tags:
+      - "Private user"
+      summary: "Get Information about an instance"
+      description: "Get the current state of an instance, this includes all events which have happened."
+      parameters:
+      - $ref: '#/parameters/instance_id'
+      produces:
+      - "application/json"
+      security:
+      - FlorenceAPIKey: []
+      responses:
+        200:
+          description: "Return a single instance state"
+          schema:
+            $ref: '#/definitions/Instance'
+        401:
+          $ref: '#/responses/UnauthorisedError'
+        403:
+          $ref: '#/responses/ForbiddenError'
+        404:
+          $ref: '#/responses/InstanceNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    put:
+      tags:
+      - "Private"
+      summary: "Update an instances properties"
+      description: "Update an instance by providing an unique id and a set of properties to over write"
+      parameters:
+      - $ref: '#/parameters/instance_id'
+      - $ref: '#/parameters/instance'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The instance has been updated"
+        401:
+          $ref: '#/responses/UnauthorisedError'
+        403:
+          $ref: '#/responses/ForbiddenError'
+        404:
+          $ref: '#/responses/InstanceNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /instances/{instance_id}/dimensions:
+    get:
+      tags:
+      - "Private user"
+      summary: "Get all dimensions from an instance"
+      description: "Get all dimensions from an instance"
+      parameters:
+      - $ref: '#/parameters/instance_id'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "Return a list of dimensions"
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/DimensionNode'
+        401:
+          $ref: '#/responses/UnauthorisedError'
+        403:
+          $ref: '#/responses/ForbiddenError'
+        404:
+          $ref: '#/responses/InstanceNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /instances/{instance_id}/dimensions/{dimension}/options:
+    get:
+      tags:
+      - "Private user"
+      summary: "Get all unique values from a dimension"
+      description: "Get all unique values from a dimension"
+      parameters:
+      - $ref: '#/parameters/instance_id'
+      - $ref: '#/parameters/dimension'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "Return a list of unique values"
+          schema:
+            type: object
+            properties:
+              dimension_id:
+                description: "The name of the dimension"
+                type: string
+              values:
```

Comment:
Given that there is an endpoint to PUT /options/{value}/node_id/{id} - the list of options should probably include the node_ids? Otherwise I dont see where we'll ever return that info